lib/commons/dom/find-pseudo-element.js

```diff
@@ -0,0 +1,71 @@
+import memoize from '../../core/utils/memoize';
+import getOwnBackgroundColor from '../color/get-own-background-color';
+
+/**
+ * Find the pseudo element of node that meets the minimum size threshold.
+ * @method findPseudoElement
+ * @memberof axe.commons.dom
+ * @instance
+ * @param {VirtualNode} vNode The VirtualNode to find pseudo elements for
+ * @param {Object} options
+ * @returns {VirtualNode|undefined} The VirtualNode which has matching pseudo elements.
+ */
+export default function findPseudoElement(
+  vNode,
+  { pseudoSizeThreshold = 0.25, ignorePseudo = false, recurse = true } = {}
+) {
+  if (ignorePseudo) {
+    return;
+  }
+  const rect = vNode.boundingClientRect;
+  const minimumSize = rect.width * rect.height * pseudoSizeThreshold;
+  do {
+    const beforeSize = getPseudoElementArea(vNode.actualNode, ':before');
+    const afterSize = getPseudoElementArea(vNode.actualNode, ':after');
+    if (beforeSize + afterSize > minimumSize) {
+      return vNode; // Combined area of before and after exceeds the minimum size
+    }
+  } while (recurse && (vNode = vNode.parent));
+}
+
+const getPseudoElementArea = memoize(function getPseudoElementArea(
+  node,
+  pseudo
+) {
+  const style = window.getComputedStyle(node, pseudo);
+  const matchPseudoStyle = (prop, value) =>
+    style.getPropertyValue(prop) === value;
+  if (
+    matchPseudoStyle('content', 'none') ||
+    matchPseudoStyle('display', 'none') ||
+    matchPseudoStyle('visibility', 'hidden') ||
+    matchPseudoStyle('position', 'absolute') === false
```

Comment:
This line works well enough when this was part of the color-contrast-evaluate method, but I'm not so sure about this now that it's in the commons.